lib/explorer/query.ex

```diff
@@ -599,5 +599,30 @@ defmodule Explorer.Query do
     |> Enum.map(&%{Explorer.Shared.apply_impl(df, :pull, [&1]) | name: &1})
   end
 
+  defmacro select(do: clauses) do
+    conditions =
+      clauses
+      |> Enum.map(fn {:->, _, [[condition], on_true]} -> [condition, on_true] end)
```

Comment:
You can do the reverse here, so we don't do it at runtime, and you can use tuples (semantically better):

```suggestion
      |> Enum.map(fn {:->, _, [[condition], on_true]} -> {condition, on_true} end)
      |> Enum.reverse()
```